### PR TITLE
Protect against NPE in linting config.

### LINKED ISSUE #88: Ability to turn off linting broke running tests
With 0.9.6, I'm now getting this error with ember-cli 0.2.7:

```
Cannot read property 'ember-cli-mocha' of undefined
TypeError: Cannot read property 'ember-cli-mocha' of undefined
  at Class.module.exports.lintTree (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli-mocha/index.js:125:21)
  at /Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:382:25
  at Array.forEach (native)
  at EmberApp.addonLintTree (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:380:23)
  at Class.Addon.jshintAddonTree (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/models/addon.js:525:32)
  at Class.treeFor (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/models/addon.js:293:21)
  at EmberApp.<anonymous> (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:342:20)
  at Array.map (native)
  at EmberApp.addonTreesFor (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:340:30)
  at EmberApp._processedAppTree (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:530:25)
  at EmberApp.appAndDependencies (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:807:24)
  at EmberApp.javascript (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:882:34)
  at EmberApp.toArray (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:1250:10)
  at EmberApp.toTree (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/broccoli/ember-app.js:1272:30)
  at Object.<anonymous> (/Users/blimmer/code/webapps/my-app/Brocfile.js:110:22)
  at Module._compile (module.js:460:26)
  at Object.Module._extensions..js (module.js:478:10)
  at Module.load (module.js:355:32)
  at Function.Module._load (module.js:310:12)
  at Module.require (module.js:365:17)
  at require (module.js:384:17)
  at Object.loadBrocfile (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/broccoli/lib/builder.js:171:14)
  at Class.module.exports.Task.extend.setupBroccoliBuilder (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/models/builder.js:21:29)
  at Class.module.exports.Task.extend.init (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/models/builder.js:35:10)
  at new Class (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/core-object/core-object.js:18:12)
  at Class.module.exports.Task.extend.run (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/tasks/build.js:15:19)
  at Class.module.exports.Command.extend.run (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/commands/test.js:118:20)
  at Class.<anonymous> (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/lib/models/command.js:136:17)
  at lib$rsvp$$internal$$tryCatch (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:493:16)
  at lib$rsvp$$internal$$invokeCallback (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:505:17)
  at lib$rsvp$$internal$$publish (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:476:11)
  at lib$rsvp$asap$$flush (/Users/blimmer/code/webapps/my-app/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1198:9)
  at process._tickCallback (node.js:355:11)
``

### PR DESCRIPTION
**Pull request by blimmer**
Fixes #88.

### CODE REVIEW

**Comment by blimmer** [comment]
@rwjblue this was a blocker for us and necessitates 0.9.7 (IMO)

**Comment by rwjblue** [comment]
Adding the guard is good, but can you also add [this](https://github.com/ember-cli/ember-cli-qunit/blob/master/index.js#L38-L41) so that `this.options` can be defined?

**Comment by blimmer** [comment]
Added!

File: index.js [hunk 1]
```diff
@@ -63,8 +63,11 @@ module.exports = {
     return path.join(__dirname, 'blueprints');
   },
 
-  included: function(app) {
-    this._super.included(app);
+  included: function included(app, parentAddon) {
+    var target = (parentAddon || app);
+    this._super.included(target);
```
Comment (rwjblue) [inline index.js]: Sorry for the annoyance, can you tweak this to `this._super.included.call(this, target)`?

Comment (blimmer) [inline index.js]: No problem! done.

**Comment by rwjblue** [comment]
Released as v0.9.7.

**Pull request merged**